rust_src/src/multibyte.rs

```diff
@@ -87,6 +87,14 @@ impl LispStringRef {
         self.data as *mut c_char
     }
 
+    pub fn const_data_ptr(&self) -> *const c_uchar {
```

Comment:
The value added by these functions in my mind isn't the return value, but in the argument. In order to get the data pointer currently, I **need** to do a mutable borrow. This would mean that in any case that I need to access data, I need mutability. This would translate into us making a lot of variables mutable that don't need to be, and could make structuring code awkward when the borrow checker enforces 'only one mutable borrow at a time'.

For example, if I use 'data_ptr', the variable lispstr must now be marked mutable, for no other reason except to satisfy data_ptr's need to do a mutable borrow.